src/v/cluster/shard_placement_table.h

```diff
@@ -140,14 +140,23 @@ class shard_placement_table
         std::optional<ss::shard_id> _next;
     };
 
+    using ntp2state_t = absl::node_hash_map<model::ntp, placement_state>;
+
     explicit shard_placement_table(storage::kvstore&);
 
     /// Must be called on assignment_shard_id.
     bool is_persistence_enabled() const;
     ss::future<> enable_persistence();
 
-    // must be called on each shard
-    ss::future<> initialize(const topic_table&, model::node_id self);
+    /// Must be called on assignment_shard_id.
+    /// precondition: is_persistence_enabled() == true
+    ss::future<> initialize_from_kvstore(
+      const chunked_hash_map<raft::group_id, model::ntp>& local_group2ntp);
```

Comment:
it seems that a comment on the `shard_placement_table` can now be updated as it is using kv_store persistence